internal/manifest/cr_deletion_check.go

```diff
@@ -0,0 +1,50 @@
+package manifest
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
+	declarative "github.com/kyma-project/lifecycle-manager/internal/declarative/v2"
+	"github.com/kyma-project/lifecycle-manager/pkg/util"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// NewModuleCRDeletionCheck creates a check that verifies that the Resource CR in the remote cluster is deleted.
+func NewModuleCRDeletionCheck() *ModuleCRDeletionCheck {
+	return &ModuleCRDeletionCheck{}
+}
+
+type ModuleCRDeletionCheck struct{}
+
+func (c *ModuleCRDeletionCheck) Run(
+	ctx context.Context,
+	clnt client.Client,
+	obj declarative.Object,
+) (bool, error) {
+	manifest, ok := obj.(*v1beta2.Manifest)
+	if !ok {
+		return false, v1beta2.ErrTypeAssertManifest
+	}
+	if manifest.Spec.Resource == nil {
+		return true, nil
+	}
+
+	name := manifest.Spec.Resource.GetName()
+	namespace := manifest.Spec.Resource.GetNamespace()
+	gvk := manifest.Spec.Resource.GroupVersionKind()
+
+	resourceCR := &unstructured.Unstructured{Object: map[string]interface{}{
```

Comment:
you can use `resourceCR.SetGroupVersionKind(gvk)` afterwards, instead of defining it as an untyped Object.

Reply:
if you find other places in code base also define gvk like this, you can also refactor in this PR.